### PR TITLE
Bug fix preapared statement close

### PR DESCRIPTION
**Pull request by afthabvp**
still need to refactor the methods which are **bulkUpsertAndReturnOlderDocuments** and **search** in **PostgresCollection.class**
facing issues describe in [https://stackoverflow.com/questions/1870022/java-iterator-backed-by-a-resultset](url)

### CODE REVIEW

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java [hunk 1]
```diff
@@ -183,7 +181,7 @@ public boolean updateSubDoc(Key key, String subDocPath, Document subDocument) {
       searchSQLBuilder.append(" OFFSET ").append(offset);
     }
 
-    try {
+    try{
```
Comment (buchi-busireddy) [inline document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java:184]: nit: Space after `try`

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java [hunk 2]
```diff
@@ -430,52 +422,53 @@ public boolean bulkUpsert(Map<Key, Document> documents) {
...
   }
 
   @Override
   public Iterator<Document> bulkUpsertAndReturnOlderDocuments(Map<Key, Document> documents) throws IOException {
+    String query = null;
     try {
       String collect = documents.keySet().stream()
-          .map(val -> "'" + val.toString() + "'")
-          .collect(Collectors.joining(", "));
+              .map(val -> "'" + val.toString() + "'")
```
Comment (buchi-busireddy) [inline document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java:448]: nit: The new indentation seems like is 8 spaces but we follow Google's Java coding style and only 4 space when the statement extends onto the new line.

**Comment by buchi-busireddy** [comment]
Thanks for this contribution @afthabvp

**Comment by jcchavezs** [comment]
Thanks @afthabvp I am merging as it as and later on add the formatting fixes and a GH action to check it.

**Review by jcchavezs — APPROVED** [review]

**Review by kotharironak — APPROVED** [review]
lgtm

**Pull request merged**